ansible/configure_galaxy.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+import argparse
+import fnmatch
+import json
+import logging
+import os
+import re
+import sys
+
+import bioblend.galaxy as bg
+
+DefaultGalaxyUrl = "http://127.0.0.1:8080"
+
+LogFormat = '%(levelname)s\t[ %(name)s ]\t%(message)s'
+Log = logging.getLogger('ConfigGalaxy')
+
+def parse_args(args):
+    parser = argparse.ArgumentParser(
+        description="Upload the workflows from a directory into the specified Galaxy server")
+
+    parser.add_argument('api_key', help="Master API key for Galaxy")
+    parser.add_argument('username', help="Username for admin user")
+    parser.add_argument('user_email', help="E-mail for admin user")
+    parser.add_argument('user_password', help="Password for admin user (at least 6 characters)")
+
+    parser.add_argument('--galaxy-url',
+                        help="Galaxy URL (default: {})".format(DefaultGalaxyUrl),
+                        default=DefaultGalaxyUrl)
+
+    parser.add_argument('--workflow-dir',
+                        help="Directory containing workflow files to be uploaded (must have .ga extension; default: '.')",
+                        default='.')
+
+    options = parser.parse_args(args)
+
+    Log.debug("Galaxy config running from %s", os.path.abspath('.'))
+    if len(options.user_password) < 6:
+        parser.error("Password needs to be at least 6 characters long.  User not added!")
+    if not re.match(r'[^@]+@[^@]+\.[^@]+', options.user_email):
+        parser.error("Email address has an invalid syntax.  User not added!")
+    if not os.path.isdir(options.workflow_dir) or not os.access(options.workflow_dir, os.R_OK | os.X_OK):
```

Comment:
I understand, thanks for the clear explanation, I think that your proposed approach makes sense given the restrictions. Maybe instead of adding the extra argument for skipping, you could simply avoid failing and not calling the workflow part if the workflow argument is missing, instead of having to actively tell this to skip the step, but whatever you decide there I'm sure it will be fine. Thanks!